src/server/snap.rs

```diff
@@ -150,32 +152,52 @@ impl<R: RaftStoreRouter + 'static> Runnable<Task> for Runner<R> {
             Task::Register(token, meta) => {
                 SNAP_TASK_COUNTER.with_label_values(&["register"]).inc();
                 let mgr = self.snap_mgr.clone();
-                match SnapKey::from_snap(meta.get_message().get_snapshot())
-                    .and_then(|key| mgr.rl().get_snap_file(&key, false).map(|r| (r, key))) {
-                    Ok((f, k)) => {
-                        if f.exists() {
-                            info!("file {} already exists, skip receiving.",
-                                  f.path().display());
-                            if let Err(e) = self.raft_router.send_raft_msg(meta) {
-                                error!("send snapshot for token {:?} err {:?}", token, e);
-                            }
-                            self.close(token);
-                            return;
-                        }
-                        debug!("begin to receive snap {:?}", meta);
-                        mgr.wl().register(k.clone(), SnapEntry::Receiving);
-                        self.files.insert(token, (f, meta));
+                let key = match SnapKey::from_snap(meta.get_message().get_snapshot()) {
```

Comment:
I prefer to shorten the indentation levels, and highlight the error handling on each step, so that the main procedure steps keep in the first indentation level.